cpp/src/arrow/filesystem/test_util.h

```diff
@@ -170,7 +170,7 @@ class ARROW_TESTING_EXPORT GenericFileSystemTest {
   virtual bool allow_move_dir_over_non_empty_dir() const { return false; }
   // - Whether the filesystem allows appending to a file
   virtual bool allow_append_to_file() const { return true; }
-  // - Whether the filesystem allows appending to a new (not existent yet) file
+  // - Whether the filesystem allows appending to a new (not yet extant) file
```

Comment:
Is this change correct?

Reply:
fwiw, https://www.merriam-webster.com/dictionary/existent seen around 1535, and 
https://www.merriam-webster.com/dictionary/extant seen around 1545.

In the context, there's a fun word that I suspect you haven't encountered: https://www.merriam-webster.com/dictionary/nascent that can be seen around 1624. It can't be used here because it'd only apply once the file exists whereas we're talking about a file that doesn't yet exist :).

The simplest wording is this:
```suggestion
  // - Whether the filesystem allows appending to a nonexistent file
```
It isn't _quite_ the same meaning as what is currently there, but it's technically what we're talking about.

Reply:
I agree with the last suggestion "appending to a nonexistent file".